src/base/bittorrent/torrenthandle.h

```diff
@@ -445,14 +445,26 @@ namespace BitTorrent
 
         QHash<QString, TrackerInfo> m_trackerInfos;
 
-        enum StartupState
+        enum ProcessingState
         {
-            NotStarted,
-            Starting,
-            Started
+            PreparingPaused, // torrent is preparing to start processing in "paused" state
+            PreparingResumed, // torrent is preparing to start processing in "resumed" state
+            Starting, // torrent is prepared and starting to perform regular processing
+            Working, // torrent is performing regular processing
+            CheckingPaused // torrent is checking in "paused" state
         };
+        // Handle "extra" states of torrent processing.
+        // Initially torrent can be either in "PreparingPaused" or "PreparingResumed" state.
+        // Torrent is become "paused" after checking to perform all needed preparations.
```

Comment:
where comes the "checking"? So far you only mentioned `PreparingPaused` & `PreparingResumed`.